src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1256,31 +1263,36 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
         }
 
         // Returns true if the status code is one of the supported redirection codes.
-        private static bool IsRedirectCode(HttpStatusCode code)
+        private static bool IsRedirectCode(HttpStatusCode statusCode)
         {
-            int intCode = (int)code;
-            return
-            (
-                (intCode >= 300 && intCode < 304) ||
-                intCode == 307 ||
-                intCode == 308
-            );
+            switch (statusCode)
+            {
+                case HttpStatusCode.Moved:
+                case HttpStatusCode.Found:
+                case HttpStatusCode.SeeOther:
+                case HttpStatusCode.TemporaryRedirect:
+                case HttpStatusCode.MultipleChoices:
+                case HttpStatusCode.PermanentRedirect:
+                    return true;
+                default:
+                    return false;
+            }
         }
 
-        // Returns true if the status code is a redirection code and the action requires switching from POST to GET on redirection.
-        // NOTE: Some of these status codes map to the same underlying value but spelling them out for completeness.
-        private static bool IsRedirectToGet(HttpStatusCode code)
+        // Returns true if the status code is a redirection code and the action requires switching to GET on redirection.
+        private static bool RequestRequiresForceGet(HttpStatusCode statusCode, WebRequestMethod requestMethod)
         {
-            return
-            (
-                code == HttpStatusCode.Found ||
-                code == HttpStatusCode.Moved ||
-                code == HttpStatusCode.Redirect ||
-                code == HttpStatusCode.RedirectMethod ||
-                code == HttpStatusCode.SeeOther ||
-                code == HttpStatusCode.Ambiguous ||
-                code == HttpStatusCode.MultipleChoices
-            );
+            switch (statusCode)
+            {
+                case HttpStatusCode.Moved:
+                case HttpStatusCode.Found:
+                case HttpStatusCode.MultipleChoices:
+                    return requestMethod == WebRequestMethod.Post;
+                case HttpStatusCode.SeeOther:
+                    return requestMethod != WebRequestMethod.Get && requestMethod != WebRequestMethod.Head;
+                default:
+                    return false;
```

Comment:
It is the same as https://source.dot.net/#System.Net.Http/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs,141
This is used here https://source.dot.net/#System.Net.Http/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs,495

It seems SocketsHttpHandler already implemented the behavior, doesn't it?

Reply:
Yes, the only difference is SocketsHttpHandler uses `HttpMethod requestMethod` and we use `WebRequestMethod requestMethod`.

Reply:
We have `private HttpMethod GetHttpMethod(WebRequestMethod method)` for the conversion.

Reply:
We need `RequestRequiresForceGet` for the cases when we set `handler.AllowAutoRedirect = false` and `WebRequestPSCmdlet` handles the redirect

Reply:
I see. Thanks for clarify!